### PR TITLE
Fix OutOfMemory error when loading game state after already having a game loaded

### PR DESCRIPTION
**Pull request by Azzurite**
Before, when we already had a game running and loaded another game, we used (more than) double the memory because we created a new `WorldScreen` before getting rid of the previous `WorldScreen`.

Now, the previous `WorldScreen` is disposed of before loading the new one. This should mean that any game you can load initially, you can also keep playing.

### CODE REVIEW

**Comment by yairm210** [comment]
I have to say I don't exactly understand which parts are supposed to do what in this change
I hope that this doesn't dispose the current world screen upon "next turn"?

File: core/src/com/unciv/ui/worldscreen/WorldScreen.kt [hunk 1]
```diff
@@ -604,26 +611,38 @@ class WorldScreen(val gameInfo: GameInfo, val viewingCiv:CivilizationInfo) : Bas
...
+            mapHolder.scrollY = restoreState.scrollY
+            mapHolder.updateVisualScroll()
         }
 
-        newWorldScreen.selectedCiv = gameInfo.getCivilization(selectedCiv.civName)
-        newWorldScreen.fogOfWar = fogOfWar
-
-        return newWorldScreen
+        selectedCiv = gameInfo.getCivilization(restoreState.selectedCivName)
+        fogOfWar = restoreState.fogOfWar
```
Comment (yairm210) [inline core/src/com/unciv/ui/worldscreen/WorldScreen.kt:637]: I'm pretty sure these are calculable from the game info, not sure if this convenience is more confusing than it is helpful

File: core/src/com/unciv/ui/worldscreen/WorldScreen.kt [hunk 2]
```diff
@@ -897,4 +894,32 @@ class WorldScreen(val gameInfo: GameInfo, val viewingCiv:CivilizationInfo) : Bas
...
+}
+
+/** This exists so that no reference to the current world screen remains, so the old world screen can get garbage collected during [UncivGame.loadGame]. */
+private fun startNewScreenJob(gameInfo: GameInfo) {
+    Concurrency.run {
+        val newWorldScreen = try {
+            UncivGame.Current.loadGame(gameInfo)
+        } catch (oom: OutOfMemoryError) {
+            val mainMenu = UncivGame.Current.goToMainMenu()
+            ToastPopup("Not enough memory on phone to load game!", mainMenu)
```
Comment (yairm210) [inline core/src/com/unciv/ui/worldscreen/WorldScreen.kt:908]: Should be in GL context

File: core/src/com/unciv/ui/LoadingScreen.kt [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+import com.badlogic.gdx.graphics.g2d.SpriteBatch
 import com.unciv.Constants
+import com.unciv.ui.popup.Popup
 import com.unciv.ui.utils.BaseScreen
-import com.unciv.ui.utils.extensions.center
-import com.unciv.ui.utils.extensions.toLabel
 
+/** A loading screen that creates a screenshot of the current screen and adds a "Loading..." popup on top of that */
 class LoadingScreen : BaseScreen() {
+    val screenshot = Texture(Pixmap.createFromFrameBuffer(0, 0, Gdx.graphics.backBufferWidth, Gdx.graphics.backBufferHeight))
```
Comment (yairm210) [inline core/src/com/unciv/ui/LoadingScreen.kt]: NEATO

Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/LoadingScreen.kt]: Why is a screenshot even necessary? The effect looks a bit strange - you get the "Loading..." text first in a _wide_ blue rectangle for a while, then in a _narrow_ blue rectangle for another second before the result pops in.

**Pull request merged**